flipperdevices/flipperzero-good-faps
# [FL-3357] Magic Gen2 support

# What's new

## Important!
**Merge https://github.com/flipperdevices/flipperzero-firmware/pull/3567 before this PR**

- Support for writing and wiping Mifare Classic and Magic Gen2 cards
- UI design WIP

# Verification 

 - For testing, using a Gen4 card locked with a password is recommended to simplify wiping and avoid wasting regular cards because of locked sectors.
- Try writing a dump to a Mifare Classic (non-magic) card, check that it has been written correctly (except for block 0)
- Try wiping that card, verify that it had been wiped (except for the locked sectors)
- Do the same with a Gen2 card or a Gen4 configured via proxmark using commands:
  - `hf 14a raw -s -c -t 1000 CF00000000CF00` - enable DirectWrite
  - `script run hf_mf_ultimatecard -z 090978009102DABC1910` - change ATS to match a Gen2 card
- Each of the cards should be detected as the correct type. However, some Gen2 cards have no ATS answer, and therefore are detected as regular Mifare Classic. This should not affect writing to block 0.

# Checklist (For Reviewer)
